### ISSUE TITLE
How can I increase the network structure?

### ISSUE THREAD

**Issue by Kobeee0614** [tkipf/gcn#88]
How can I increase the network structure? I have thought about operating in the _build() function in class GCN. Is this correct?

**Comment by tkipf** [comment]
Could you please clarify what you mean by "increasing the network
structure"? Do you mean adding more convolutional/fully-connected layers to
the GCN? In this case, adapting the `_build()` method of the GCN module is
the correct thing to do.

**Comment by Kobeee0614** [comment]
Yes, you are absolutely right. But I found I increase the convolutional/fully - connected the layers, the network performance does not improve, I don't know where I went wrong, can you help me see? Also, if I delete the network structure in the _build() function, it still works?`self.layers.append(GraphConvolution(input_dim=self.input_dim,
                                            output_dim=FLAGS.hidden1,
                                            placeholders=self.placeholders,
                                            act=tf.nn.relu,
                                            dropout=True,
                                            sparse_inputs=True,
                                            logging=self.logging))


        self.layers.append(GraphConvolution(input_dim=FLAGS.hidden1,
                                            output_dim=FLAGS.hidden2,
                                            placeholders=self.placeholders,
                                            act=lambda x: x,
                                            dropout=True,
                                            logging=self.logging))

        self.layers.append(GraphConvolution(input_dim=FLAGS.hidden2,
                                            output_dim=FLAGS.hidden3,
                                            placeholders=self.placeholders,
                                            act=lambda x: x,
                                            dropout=True,
                                            logging=self.logging))

        self.layers.append(GraphConvolution(input_dim=FLAGS.hidden3,
                                            output_dim=FLAGS.hidden4,
                                            placeholders=self.placeholders,
                                            act=lambda x: x,
                                            dropout=True,
                                            logging=self.logging))

        self.layers.append(GraphConvolution(input_dim=FLAGS.hidden4,
                                            output_dim=self.output_dim,
                                            placeholders=self.placeholders,
                                            act=lambda x: x,
                                            dropout=True,
                                            logging=self.logging))`

**Comment by tkipf** [comment]
Adding more layers doesn't necessarily improve performance (see discussion
in Appendix here: https://arxiv.org/abs/1609.02907). Also, you did not
place any activation functions on your layers, so everything essentially
reduces to a linear model. Try ReLU activations, for example (for all but
the last layer).
